tournamentcontrol/competition/signals/matches.py

```diff
@@ -150,3 +155,164 @@ def notify_match_forfeit_email(sender, match, team, *args, **kwargs):
+        'competition:match-video',
+        kwargs={
+            'competition': competition.slug,
+            'season': season.slug,
+            'division': division.slug,
+            'match': obj.pk,
+        }
+    )
+    match_url = f"https://{current_site.domain}{match_path}"
+    
+    if obj.play_at and hasattr(obj.play_at, 'ground') and obj.play_at.ground.venue:
+        venue_text = f"from {obj.play_at.ground.venue}"
+        play_at_text = f"on {obj.play_at}"
+    else:
+        venue_text = ""
+        play_at_text = ""
+    
+    description = (
+        f"Live stream of the {division} division of {competition} {season} "
+        f"{venue_text}.\n"
+        f"\n"
+        f"Watch {obj.get_home_team_plain()} take on "
+        f"{obj.get_away_team_plain()} {play_at_text}.\n"
+        f"\n"
+        f"Full match details are available at {match_url}\n"
+        f"\n"
+        f"Subscribe to receive notifications of upcoming matches."
+    )
+
+    start_time = obj.get_datetime(ZoneInfo("UTC"))
+    stop_time = obj.get_datetime(ZoneInfo("UTC")) + relativedelta(
+        minutes=50
+    )  # FIXME: hard coded
+
+    body = {
+        "snippet": {
+            "title": title,
+            "description": description,
+            "scheduledStartTime": start_time.isoformat(),
+            "scheduledEndTime": stop_time.isoformat(),
+        },
+        "status": {
+            "privacyStatus": season.live_stream_privacy,
+            "selfDeclaredMadeForKids": False,
+        },
+        "contentDetails": {
+            "enableAutoStart": False,
+            "enableAutoStop": False,
+            "monitorStream": {
+                "broadcastStreamDelayMs": 0,
+                "enableMonitorStream": True,
+            },
+        },
+    }
+
+    try:
+        if obj.external_identifier:
+            # If we have disabled live-streaming where it was previously
+            # enabled, we need to remove it using the YouTube API.
+            if not obj.live_stream:
+                video_id = obj.external_identifier
+                season.youtube.liveBroadcasts().delete(id=video_id).execute()
+                if obj.videos is not None:
+                    obj.videos.remove(f"https://youtu.be/{video_id}")
+                if not obj.videos:
+                    obj.videos = None
+                obj.external_identifier = None
+                logger.info("YouTube video %r deleted", video_id)
+                return
+
+            # Alternatively we're making sure the representation on the backend
+            # is consistent with the current status.
+            else:
+                body["id"] = obj.external_identifier
+                broadcast = (
+                    season.youtube.liveBroadcasts()
+                    .update(part="snippet,status,contentDetails", body=body)
+                    .execute()
+                )
+                from tournamentcontrol.competition.tasks import set_youtube_thumbnail
```

Comment:
This `import` should be at the top of the file unless it is required to avoid a cycle.